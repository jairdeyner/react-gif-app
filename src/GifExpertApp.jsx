import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Goku"]);

  const onAddCategory = (newCategory) => {
    if (categories.some((c) => c.toLowerCase() === newCategory.toLowerCase()))
      return;

    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GifExpertAppp</h1>

      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((c) => (
        <GifGrid key={c} category={c} />
      ))}
    </>
  );
};
