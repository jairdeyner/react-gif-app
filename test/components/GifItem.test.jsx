import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("pruebas en <GifItem />", () => {
  const title = "Goku";
  const url = "https://goku.com/goku.png";

  test("debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);

    const $img = screen.getByRole("img");

    expect($img.src).toBe(url);
    expect($img.alt).toBe(title);
  });

  test("debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
