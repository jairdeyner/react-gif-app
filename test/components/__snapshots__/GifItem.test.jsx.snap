// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://goku.com/goku.png"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
